File: typescript/src/com/murex/tbw/domain/country/Country.ts
import { Currency } from "./Currency";
import { Language } from "./Language";

export class Country {
  readonly name: string;
  readonly currency: Currency;
  readonly language: Language;

  public constructor(name: string, currency: Currency, language: Language) {
    this.name = name;
    this.currency = currency;
    this.language = language;
  }
}

File: fsharp/Application/Domain/Country.fs
namespace Application.Domain

type Currency = 
    | UsDollar
    | AustralianDollar
    | Euro
    | PoundSterling
    | Yen
    | Reminbi

type Language =
    | English
    | French
    | Spanish
    | Japanese
    | Mandarin
    | Arabic
    | German

type CountryName =
    | USA
    | France
    | UK
    | Spain
    | China
    | Japan
    | Australia
    | Germany

type Country = {
    Name: CountryName
    Currency: Currency
    Language: Language
}

